shared/src/IWritablePropertyResponse.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Shared
+{
+    /// <summary>
+    /// The interface that defines the structure of a writable property response.
+    /// </summary>
+    /// <remarks>
+    /// This interface is used to allow extension to use a different set of attributes for serialization.
+    /// For example our default implementation found in <see cref="NewtonsoftJsonWritablePropertyResponse"/> is based on <see cref="Newtonsoft.Json"/> seriailizer attributes.
+    /// </remarks>
+    public interface IWritablePropertyResponse
+    {
+        /// <summary>
+        /// The unserialized property value.
+        /// </summary>
+        public object Value { get; set; }
```

Comment:
I see value in calling this propertyValue. Thoughts?

Reply:
Since the name calls out that this is a "writable property" response, we didn't call it out again in the property (.NET property) names.
Also, the .NET property names are aligned to be as close as possible (while still understandable) to the pnp-json-property names (value, ac, av,ad)